packages/attention/src/component.tsx

```diff
@@ -1,154 +1,237 @@
+	const attentionRef = useRef<HTMLDivElement | null>(null)
+	const arrowRef = useRef<HTMLDivElement | null>(null)
+
+	const attentionState = {
+		get isShowing() {
+			return isShowing
+		},
+		get isCallout() {
+			return rest.callout
+		},
+		get actualDirection() {
+			return actualDirection
+		},
+		set actualDirection(v) {
+			setActualDirection(v)
+		},
+		get directionName() {
+			return placement
+		},
+		get arrowEl() {
+			return arrowRef.current
+		},
+		get attentionEl() {
+			return attentionRef.current
+		},
+		set attentionEl(v) {
+			attentionRef.current = v
+		},
+		get targetEl() {
+			return targetEl?.current
+		},
+		get noArrow() {
+			return props.noArrow
+		},
+	}
+
+	//TODO: See if we can move this function to the core-repo:
+	const translatedDirection = (() => {
+		switch (opposites[actualDirection]) {
+			case 'top':
+				return i18n._({
+					id: 'attention.direction.top',
+					message: 'top',
+					comment:
+						'Default screenreader message for top direction in the attention component',
+				})
+			case 'right':
+				return i18n._({
+					id: 'attention.direction.right',
+					message: 'right',
+					comment:
+						'Default screenreader message for right direction in the attention component',
+				})
+			case 'bottom':
+				return i18n._({
+					id: 'attention.direction.bottom',
+					message: 'bottom',
+					comment:
+						'Default screenreader message for bottom direction in the attention component',
+				})
+			case 'left':
+				return i18n._({
+					id: 'attention.direction.left',
+					message: 'left',
+					comment:
+						'Default screenreader message for left direction in the attention component',
+				})
+			default:
+				return ''
+		}
+	})()
+
+	// TODO: See if we can move this function to the core repo:
+	const activeAttentionType = (() => {
+		switch (true) {
+			case props.tooltip:
+				return i18n._({
+					id: 'attention.tooltip',
+					message: 'tooltip',
+					comment:
+						'Default screenreader message for tooltip in the attention component',
+				})
+			case props.callout:
+				return i18n._({
+					id: 'attention.callout',
+					message: 'callout',
+					comment:
+						'Default screenreader message for callout in the attention component',
+				})
+			case props.popover:
+				return i18n._({
+					id: 'attention.popover',
+					message: 'popover',
+					comment:
+						'Default screenreader message for popover in the attention component',
+				})
+			default:
+				return ''
+		}
+	})()
+
+	// TODO: See if we can move this function to the core repo:
+	const getDefaultAriaLabel = () => {
+		const pointingAtTranslation = i18n._({
+			id: 'attention.ariaLabel.pointingAt',
+			message: 'pointing to the',
+			comment:
+				'Default screenreader message for the pointingAt function in the attention component',
+		})
+
+		const pointingAt = !props.noArrow
+			? `${pointingAtTranslation} ${translatedDirection}`
+			: ''
+
+		return `${activeAttentionType} ${pointingAt}`
```

Comment:
Maybe this could be part of the `translatedDirection` constant (it could then be called e.g. `pointingAtDirection` or sth)? Then instead of having:
```
case 'top':
  return .... i18n._({ ... message: 'top', ...}}
```
we'd do:
```
case 'top':
  return .... i18n._({ ... message: ' pointing up', ...}}
```
This would allow us to have this in line 157:
```
return `${activeAttentionType}${props.noArrow ? '' : pointingAtDirection}`
```

Wdyt?

Reply:
Sure! Sounds good, I'll change it! 😄 👍